code/modules/mob/living/carbon/xenomorph/castes/warrior/warrior.dm

```diff
@@ -75,3 +83,36 @@
 	if(ishuman(AM))
 		return
 	..()
+
+// ***************************************
+// *********** Primordial procs
+// ***************************************
+///Handles primordial warrior empowered abilities, returns TRUE if the ability should be empowered.
+/mob/living/carbon/xenomorph/warrior/proc/empower()
+	if(upgrade != XENO_UPGRADE_FOUR)
+		return FALSE
+	if(combo >= WARRIOR_COMBO_THRESHOLD) //Fully stacked, clear all the stacks and return TRUE.
+		emote("roar")
+		clear_combo()
+		return TRUE
+	give_combo()
+	return FALSE
+
+/mob/living/carbon/xenomorph/warrior/proc/give_combo() //Handles visuals aswell
+	if(upgrade != XENO_UPGRADE_FOUR)
```

Comment:
you checked that already in empower

Reply:
there is a call for combo without empowered in lunge, since I don't want it to consume combo but still give some. Hence the check.